manager/controlapi/config.go

```diff
@@ -0,0 +1,264 @@
+package controlapi
+
+import (
+	"bytes"
+	"regexp"
+	"strings"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/docker/swarmkit/api"
+	"github.com/docker/swarmkit/identity"
+	"github.com/docker/swarmkit/log"
+	"github.com/docker/swarmkit/manager/state/store"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+)
+
+// Currently this is contains the unimplemented config functions in order to satisfy the interface
+
+// MaxConfigSize is the maximum byte length of the `Config.Spec.Data` field.
+const MaxConfigSize = 500 * 1024 // 500KB
+
+var validConfigNameRegexp = regexp.MustCompile(`^[a-zA-Z0-9]+(?:[a-zA-Z0-9-_.]*[a-zA-Z0-9])?$`)
```

Comment:
Non-blocking:  this is the same for secrets, so wondering if we should just rename the secrets one to be more general and just use that one?

Would we want fewer restrictions on the config name?